desc/geometry/surface.py

```diff
@@ -1146,3 +1146,114 @@ def get_axis(self):
         data = self.compute(["R", "Z"], grid=grid)
         axis = FourierRZCurve(R_n=data["R"][0], Z_n=data["Z"][0], sym=self.sym)
         return axis
+
+
+def _constant_offset_surface(
+    base_surface, offset, grid, M, N, full_output=False, params=None
+):
+    """Create a FourierRZSurface with constant offset from the base surface (self).
+
+    Implementation of algorithm described in Appendix B of
+    "An improved current potential method for fast computation of
+    stellarator coil shapes", Landreman (2017)
+    https://iopscience.iop.org/article/10.1088/1741-4326/aa57d4
+
+    NOTE: Must have the toroidal angle as the cylindrical toroidal angle
+    in order for this algorithm to work properly
+
+    NOTE: this function lacks the checks of the constant_offset_surface
```

Comment:
only is differentiable though if params is passed, otherwise the end will not be differentiatedcorrectlt